images/ceph/Makefile

```diff
@@ -121,3 +123,15 @@ csv: $(OPERATOR_SDK) $(YQ) ## Generate a CSV file for OLM.
 
 csv-clean: $(OPERATOR_SDK) $(YQ) ## Remove existing OLM files.
 	@rm -fr ../../cluster/olm/ceph/deploy/* ../../cluster/olm/ceph/templates/*
+
+# list-image creates list of images for offline installation
+list-image:
+	@echo "producing list of images for offline installation";\
+	# remove the file if already exists
+	[ -f ../../cluster/examples/kubernetes/ceph/images.txt ] && rm ../../cluster/examples/kubernetes/ceph/images.txt;\
+	awk '/image:/ {print $2}' ../../cluster/examples/kubernetes/ceph/operator.yaml ../../cluster/examples/kubernetes/ceph/cluster.yaml | \
+	cut -d: -f2- |\
+	tee -a ../../cluster/examples/kubernetes/ceph/images.txt && \
+	awk '/quay.io/ || /k8s.gcr.io/ {print $3}' ../../cluster/examples/kubernetes/ceph/operator.yaml | \
```

Comment:
What if we read these from the code instead? It is rare but sometimes happens that the manifests don't get updated when the default images in the code are updated. The sed to find these can then look for `Default.*Image[[:whitespace:]]+=[[:whitespace:]]+"<image name>"$` (attempt at regex, but might have bugs).

From `pkg/operator/ceph/csi/spec.go`:
```
	DefaultCSIPluginImage         = "quay.io/cephcsi/cephcsi:v3.4.0"
	DefaultRegistrarImage         = "k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.2.0"
	DefaultProvisionerImage       = "k8s.gcr.io/sig-storage/csi-provisioner:v2.2.2"
	DefaultAttacherImage          = "k8s.gcr.io/sig-storage/csi-attacher:v3.2.1"
	DefaultSnapshotterImage       = "k8s.gcr.io/sig-storage/csi-snapshotter:v4.1.1"
	DefaultResizerImage           = "k8s.gcr.io/sig-storage/csi-resizer:v1.2.0"
	DefaultVolumeReplicationImage = "quay.io/csiaddons/volumereplication-operator:v0.1.0"
```

Reply:
we can read images from `pkg/operator/ceph/csi/spec.go` for some images but in the end, we have to fetch from manifest for complete images list. IMO manifests is the right place .

Reply:
For the `rook/ceph:<tag>` image and `ceph/ceph:<tag>` image, the only place they are defined is in the manifests. 

But it has happened more than once that the CSI image defined in `spec.go` is different from what is in the manifests. When (not if) that happens in the future, Rook will fail to start up for air-gapped users because they will have mirrored images that Rook isn't using by default. As far as we know, most users **do** use the internal default images, so it is paramount that these images are the ones we list in this file.

-----

There are a couple other routes we could consider, probably for a future PR. 

1. We could add images from both the manifests **and** the Go code and remove duplicate tags. We could also run an analysis on the file to see if there are images listed with multiple version, and if that's the case, we know that the manifests and Go code haven't been updated both at the same time.
2. (probably better) We could use the Go code to generate/modify the `CSI_..._IMAGE` vars that are commented-out in the manifests so they don't get out of sync.

Reply:
I see your points. so can we go ahead with this PR and create a follow-up PR?